libtenzir/src/package.cpp

```diff
@@ -236,8 +251,15 @@ auto package_config::parse(const view<record>& data)
   -> caf::expected<package_config> {
   auto result = package_config{};
   for (const auto& [key, value] : data) {
+    TRY_CONVERT_TO_STRING(version, version);
     TRY_ASSIGN_STRUCTURE_TO_RESULT(source, package_source);
     TRY_ASSIGN_STRINGMAP_CONVERSION_TO_RESULT(inputs);
+    TRY_ASSIGN_RECORD_TO_RESULT(overrides);
+    TRY_ASSIGN_RECORD_TO_RESULT(metadata);
+    return diagnostic::error("unknown key '{}'", key)
+      .note("while trying to parse 'config' entry")
+      .note("invalid package definition")
+      .to_error();
```

Comment:
```suggestion
    return diagnostic::error("unknown key `{}`", key)
      .note("while trying to parse `config` entry")
      .note("invalid package definition")
      .to_error();
```

For consistency with other diagnostics, please use backticks when referencing values or field names.